### ISSUE TITLE
privacy_engine.detach() not working?

### ISSUE THREAD

**Issue by wyshi** [pytorch/opacus#163]
## 🐛 Bug

Not able to detach the privacy_engine, optimizer not working after detach

## Please reproduce using our [template Colab]
https://colab.research.google.com/drive/15auLZ8sN4BBvP3XSjBi8hCgm172fc_T2?usp=sharing

## To Reproduce
simply run the colab, the code block that doesn't work look like this

```
optimizer.step() # works
privacy_engine.attach(optimizer)
privacy_engine.detach()
optimizer.step() # does not work
```

The error is ``TypeError: 'SGD' object is not callable``, seems like the ``original_step`` is not copied properly

## Expected behavior

expect the ``optimizer.step()`` after the ``privacy_engine.detach()`` to work as before 

## Environment

Please refer to the colab.

## Additionally, Is it possible to alternately attach and detach the privacy engine to speed things up? pseudo-code would look like

```
for i, (x, y) in enumerate(tqdm(train_loader)):
    if i % 2 == 0:
        # calculate batch gradients
        privacy_engine.detach()
    else:
        # calculate per-sample gradients
        privacy_engine.attach(optimizer)
      
    x = x.to(device)
    y = y.to(device)

    logits = model(x)
    loss = criterion(logits, y)
    loss.backward()

    optimizer.step()
    optimizer.zero_grad()

    preds = logits.argmax(-1)
    n_correct = float(preds.eq(y).sum())
    batch_accuracy = n_correct / len(y)

    accs.append(batch_accuracy)
    losses.append(float(loss))
```

**Comment by alexandresablayrolles** [comment]
Thanks for flagging the bug, we are working on a fix. I am not sure why you want to alternately attach and detach the optimizer from the engine?

**Comment by alexandresablayrolles** [comment]
Fixed by 41eb0e6678df2d79065b153608fd14dab7e5baa7
